dms/models/abstract_dms_mixin.py

```diff
@@ -0,0 +1,28 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AbstractDmsMixin(models.AbstractModel):
+    _name = "abstract.dms.mixin"
+    _description = "Abstract Dms Mixin"
+
+    res_model = fields.Char(string="Linked attachments model")
+    res_id = fields.Integer(string="Linked attachments record ID")
+    record_ref = fields.Reference(
+        string="Record Referenced",
+        compute="_compute_record_ref",
+        selection=lambda self: self._get_ref_selection(),
```

Comment:
That is the real change that is fixing this. The refactoring is just made to minimize the amount of code

Reply:
OK, understood. Then the commit message should be `[REF+FIX] ` and explaining in the commit message the real fix and the refactoring.

Reply:
Changed it now. 
Probably it's possible to optimize a lot of code that is used in some models (directory and field specially).